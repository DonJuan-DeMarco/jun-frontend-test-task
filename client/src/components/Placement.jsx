import { useSelector } from 'react-redux';

const Placement = () => {
	const order = useSelector((state) => state.horses.finishOrder);

	return (
		<div>
			<p>Finish order:</p>

			{order.map((place) => (
				<p>{place}</p>
			))}
		</div>
	);
};

export default Placement;
