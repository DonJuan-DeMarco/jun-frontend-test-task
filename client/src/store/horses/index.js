import { createSlice } from '@reduxjs/toolkit';
import { initialState } from './state';
import { reducers } from './reducers';

const horseSlice = createSlice({
	name: 'horses',
	initialState,
	reducers,
});

export const horseActions = horseSlice.actions;
export default horseSlice;
