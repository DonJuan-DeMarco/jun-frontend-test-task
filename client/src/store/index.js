import { configureStore } from '@reduxjs/toolkit';
import horseSlice from './horses';

const store = configureStore({
	reducer: {
		horses: horseSlice.reducer,
	},
});

export default store;
