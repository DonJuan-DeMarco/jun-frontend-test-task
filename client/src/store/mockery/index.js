import { createSlice } from '@reduxjs/toolkit';
import { initialState } from './state';
import { reducers } from './reducers';

const mockSlice = createSlice({
	name: 'mockery',
	initialState,
	reducers,
});

export const mockActions = mockSlice.actions;
export default mockSlice;
