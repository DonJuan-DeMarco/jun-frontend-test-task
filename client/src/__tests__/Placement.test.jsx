import { render, screen, cleanup } from '@testing-library/react';
import { Provider, useDispatch, useSelector } from 'react-redux';
import Placement from '../components/Placement';
import store from '../store';

afterEach(() => {
	cleanup();
});

test('Should render ProgressBar component', () => {
	render(
		<Provider store={store}>
			<Placement />
		</Provider>
	);
	const placementElement = screen.getByTestId('placement');
	expect(placementElement).toBeInTheDocument();
});
