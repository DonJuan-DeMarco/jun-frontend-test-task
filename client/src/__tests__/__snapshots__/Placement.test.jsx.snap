// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ProgressBar component 1`] = `
<div
  className="podium"
  data-testid="placement"
>
  <p
    className="finish"
  >
    Finish order:
  </p>
  <div
    className="place w1"
  >
    <p>
      1
    </p>
    <p>
      Princess Diana
    </p>
  </div>
  <div
    className="place w2"
  >
    <p>
      2
    </p>
    <p>
      Cricket
    </p>
  </div>
</div>
`;
